added some words to documentation LaTeX examples (#558)
ManimCommunity/manim: docs/source/examples/tex.rst

LaTeX
=================================

The Tex() mobject
+++++++++++++++++++

> Just as you can use `Text()` to add text to your videos, you can use `Tex()` to insert LaTeX.

Reviewer: ```suggestion
Just as you can use ``Text()`` to add text to your videos, you can use ``Tex()`` to insert LaTeX.
```

Author: or should that be \`:class:~.Text()\` ?

Reviewer 2: ```
:class:`~.Text`
```

Author: I went with 
```
``Text()``
```

Reviewer 2: Please use the proper reference to create a link to the `Text` class.

> Just as you can use ``Text()`` to add text to your videos, you can use ``Tex()`` to insert LaTeX.

Reviewer 2: ```suggestion
Just as you can use :class:`~.Text` to add text to your videos, you can use :class:`~.Tex` to insert LaTeX.
```

.. manim:: ExampleLaTeX
    :save_last_frame:

    class ExampleLaTeX(Scene):
        def construct(self):
            tex = Tex(r'\LaTeX').scale(3)
            self.add(tex)

> Note that we are using a raw string (r'---') instead of a regular string ('---').
> This is because TeX code uses a lot of special characters - like `\\` for example - 

Reviewer 2: ```suggestion
Note that we are using a raw string (``r'---'``) instead of a regular string (``'---'``).
This is because TeX code uses a lot of special characters - like ``\`` for example - 
```

that have special meaning within a regular python string. An alternative would have

> been to write `\\\\` as in Tex('\\\\LaTeX').

Reviewer: I think this isn't rendering the way you intended it.

Author: that's the weird thing, it is rendering correctly on my system at home...

Reviewer 2: This paragraph needs some literals (double backtick blocks) or so.

+1 for the recommendation to use raw strings.

The MathTex() mobject
++++++++++++++++++++++

> Anything enclosed in `\$` signs is interpreted as maths-mode:

Reviewer 2: ```suggestion
Anything enclosed in ``$`` signs is interpreted as maths-mode:
```

            tex = Tex(r'$\xrightarrow{Hello}$ \LaTeX').scale(3)
            self.add(tex)

> Whereas in a `MathTex()` mobject everything is math-mode by default and you would use `\\text{}` to

Reviewer: ```suggestion
Whereas in a ``MathTex()`` mobject everything is math-mode by default and you would use `\\text{}` to
```

Author: or maybe :class:\`~.MathTex()\` ?

> Whereas in a ``MathTex()`` mobject everything is math-mode by default and you would use ``\\text{}`` to

Reviewer 2: ```suggestion
Whereas in a :class:`~.MathTex` mobject everything is math-mode by default and you would use ``\text{}`` to
```

insert regular text:

.. manim:: Example1bLaTeX
    :save_last_frame:

    class Example1bLaTeX(Scene):
        def construct(self):
            tex = MathTex(r'\xrightarrow{Hello}\text{ \LaTeX}').scale(3)
            self.add(tex)

LaTeX commands and keyword arguments
+++++++++++++++++++++++++++++++++++++

> We can use any standard LaTeX commands in the AMS maths packages. For example the `mathtt` text type.

Reviewer: ```suggestion
We can use any standard LaTeX commands in the AMS maths packages. For example the ``mathtt`` text type.
```

.. manim:: Example2LaTeX
    :save_last_frame:

    class Example2LaTeX(Scene):
        def construct(self):
            tex = Tex(r'$\mathtt{Hello}$ \LaTeX').scale(3)
            self.add(tex)

> On the manim side, the TeX class also accepts attributes to change the appearance of the output. 

Reviewer 2: ```suggestion
On the manim side, the :class:`~.Tex` class also accepts attributes to change the appearance of the output. 
```

> This is very similarly to the Text class. For example, the `color` keyword changes the color of the TeX mobject:

Reviewer 2: ```suggestion
This is very similar to the :class:`~.Text` class. For example, the ``color`` keyword changes the color of the TeX mobject:
```

.. manim:: Example2bLaTeX
    :save_last_frame:

    class Example2bLaTeX(Scene):
        def construct(self):
            tex = Tex(r'Hello \LaTeX', color=BLUE).scale(3)
            self.add(tex)

Extra LaTeX Packages
+++++++++++++++++++++
Some commands require special packages to be loaded into the TeX template. For example, 

> to use the `mathscr` script, we need to add the `mathrsfs` package. Since this package isn't loaded

Reviewer: ```suggestion
to use the ``mathscr`` script, we need to add the ``mathrsfs`` package. Since this package isn't loaded
```

        def construct(self):
            myTemplate = TexTemplate()
            myTemplate.add_to_preamble(r"\usepackage{mathrsfs}")
            tex = Tex(r'$\mathscr{H} \rightarrow \mathbb{H}$}', tex_template=myTemplate).scale(3)
            self.add(tex)

Substrings and parts
+++++++++++++++++++++
The TeX mobject can accept multiple strings as arguments. Afterwards you can refer to the individual

> parts either by their index (like `tex[1]`), or you can look them up by (parts of) the tex code like

Reviewer 2: ```suggestion
parts either by their index (like ``tex[1]``), or you can look them up by (parts of) the tex code like
```

> in this example where we set the color of the `\\bigstar` using `set_color_by_tex()`:

Reviewer 2: ```suggestion
in this example where we set the color of the ``\bigstar`` using :func:`~.set_color_by_tex`:
```

.. manim:: Example4LaTeX
    :save_last_frame:

    class Example4LaTeX(Scene):
        def construct(self):
            tex = Tex('Hello', r'$\bigstar$', r'\LaTeX').scale(3)
            tex.set_color_by_tex('igsta', RED)
            self.add(tex)

LaTeX Maths Fonts - The Template Library
++++++++++++++++++++++++++++++++++++++++++++

> Changing fonts in LaTeX when typesetting mathematical formulae is a little bit more tricky than 

Reviewer 3: ```suggestion
Changing fonts in LaTeX when typesetting mathematical formulas is a little bit more tricky than 
```
From here:https://www.grammar-monster.com/plurals/plural_of_formula.htm#:~:text=Plural%20of%20Formula-,The%20plural%20of%20formula%20is%20formulas%20or%20formulae.,express%20relationships%20between%20various%20quantities.
* Formulas are applied to provide a mathematical solution for real world problems. 
* Formulae are used to express relationships between various quantities

Author: exactly.
We're not applying a mathematical solution to the real world, we're expressing a relationship between various quantities. Ergo 'formulae' is correct.

Reviewer 2: I hear for the first time that there is a semantic difference between formulas and formulae. I am not convinced that it is true; formulas and formulae are both correct plurals of 'formula' (with the 'ae'-ending coming from Latin). Merriam-Webster also does not mention a differrence in meaning: https://www.merriam-webster.com/dictionary/formulae -- but 'formulas' is certainly more common in non-scientific contexts.

Author: but on the other hand, everyone I knew in the US uses formulas as the plural and so general usage is "formulas"

Author: @kolibril13 
"Formulae" is not wrong.
"Formulas" is 'also ok', because it is widely used, but even by your quoted definition, formulae is technically better.

Reviewer 3: So what do we want to keep now?
It would be nice to have consistency so I would like to choose one. (e.g. here it is formulas https://manimce.readthedocs.io/en/latest/examples/formulas.html)
As formulas might be better known, I would tend to use that one. Actually, reviewing this pr here was the first time I heard the term Formulae. 
But in case you would prefer Formulae, we can make this the default.

Author: If you feel that strongly about it, make a PR to https://github.com/cobordism/manim/tree/docs ;)

I have always used formulae in a mathematical context, but I do not require consistency because the language use isn't consistent either. Indeed, I think the attempt by grammar-monster.com at a semantic distinction, in fact only exemplifies that "formulae" is more commonly used by scientists while "formulas" is more commonly used by the general public. Merriam-Webster would seem to concur that either is fine... and so we too can use either (and both).

Reviewer 3: >  because the language use isn't consistent either
True words! I sometimes have the tendency to try to get things as consistent as possible, however, there can be also a charming aspect of having inconsistencies, as a sign of a multi-national tinkering around to create something beautiful together, each one with their own style.
So let's keep these two versions simultaneous :)

with regular text. It requires changing the template that is used to compile the tex code.

> Manim comes with a collection of `TexFontTemplates` ready for you to use. These templates will all work

Reviewer 2: ```suggestion
Manim comes with a collection of :class:`~.TexFontTemplates` ready for you to use. These templates will all work
```

in maths mode:

.. manim:: Example5LaTeX
    :save_last_frame:

    class Example5LaTeX(Scene):
        def construct(self):
            tex = Tex(r'$f: A \rightarrow B$', tex_template=TexFontTemplates.french_cursive).scale(3)
            self.add(tex)

> Manim also has a `TexTemplateLibrary` containing the TeX templates used by 3Blue1Brown. One example 

Reviewer: ```suggestion
Manim also has a ``TexTemplateLibrary`` containing the TeX templates used by 3Blue1Brown. One example 
```

> Manim also has a ``TexTemplateLibrary`` containing the TeX templates used by 3Blue1Brown. One example 

Reviewer 2: ```suggestion
Manim also has a :class:`~.TexTemplateLibrary` containing the TeX templates used by 3Blue1Brown. One example 
```

is the ctex template, used for typesetting Chinese. For this to work, the ctex LaTeX package
must be installed on your system. Furthermore, if you are only typesetting Text, you probably do not

> need `Tex()` at all, and should use `Text()` or `PangoText()` instead. 

Reviewer 2: ```suggestion
need :class:`~.Tex` at all, and should use :class:`~.Text` or :class:`~.PangoText` instead. 
```

.. manim:: Example6LaTeX
    :save_last_frame:

    class Example6LaTeX(Scene):
        def construct(self):
            tex = Tex('Hello 你好 \\LaTeX', tex_template=TexTemplateLibrary.ctex).scale(3)
            self.add(tex)

> Aligning formulae

Reviewer 3: ```suggestion
Aligning formulas
```

++++++++++++++++++

> A `MathTex()` mobject is typeset in the LaTeX  "align" environment. This means you can use the `"&"` alignement

Reviewer 2: ```suggestion
A :class:`~.MathTex` mobject is typeset in the LaTeX  ``align*`` environment. This means you can use the ``&`` alignment
```

> character when typesetting multiline formulae:

Reviewer 3: ```suggestion
character when typesetting multiline formulas:
```

.. manim:: Example7LaTeX
    :save_last_frame:

    class Example7LaTeX(Scene):
        def construct(self):
            tex = MathTex(r'f(x) &= 3 + 2 + 1\\ &= 5 + 1 \\ &= 6').scale(2)

>             self.add(tex)

Reviewer 3: Looks great!
I think it could also go to https://manimce--558.org.readthedocs.build/en/558/examples/formulas.html

Author: yes. eventually latex and formulas should be merged into just Tex-mobject documentation...
